cli/src/complete.rs

```diff
@@ -641,6 +666,43 @@ pub fn interdiff_files(current: &std::ffi::OsStr) -> Vec<CompletionCandidate> {
     modified_files_from_rev((from, Some(to)), true, current)
 }
 
+/// Specific function for completing file paths for `jj log`
+pub fn log_files(current: &std::ffi::OsStr) -> Vec<CompletionCandidate> {
+    let Some(current) = current.to_str() else {
+        return Vec::new();
+    };
+    let rev = parse::log_revision();
+    with_jj(|mut jj, _| {
+        let output = jj
+            .arg("log")
+            .arg("--no-graph")
+            .arg("--template=")
+            .arg("--summary")
+            .arg("--revisions")
+            .arg(rev)
+            .output()
+            .map_err(user_error)?;
+        let stdout = String::from_utf8_lossy(&output.stdout);
```

Comment:
What do you mean by most recent revision? And what sparse pattern?

I do think it would've been nice if the completions included files that were deleted (/ renamed) in one revision, but were present in another. But it's a small compromise if the performance is much better.

Reply:
> What do you mean by most recent revision?

`--limit 1` or `heads(..)`. It's not uncommon to query `jj log -r::main <PATH>`, and the `<PATH>` is usually contained in the `main` revision.

> And what sparse pattern?

It's related to the stuff Martin pointed out. Listing all files can be super expensive, and it might make sense to filter paths by the working-copy's sparse pattern in addition to the current path prefix.